fiftyone/core/dataset.py

```diff
@@ -78,6 +78,32 @@ def dataset_exists(name):
     return bool(list(conn.datasets.find({"name": name}, {"_id": 1}).limit(1)))
 
 
+def _validate_dataset_name(name, skip=None):
+    """Validates that the given dataset name is available.
+
+    Args:
+        name: a dataset name
+        skip (None): an optional :class:`Dataset` to ignore
+
+    Returns:
+        the URL name
+
+    Raises:
+        ValueError: if the name is not available
+    """
+    url_name = fou.to_url_name(name)
+
+    query = {"$or": [{"name": name}, {"url_name": url_name}]}
+    if skip is not None:
```

Comment:
curious what a use case is  for skip 🤔

Reply:
So that `my dataset` can be renamed to `my-dataset` 🤗